python/lsst/ip/isr/overscan.py

```diff
@@ -0,0 +1,404 @@
+
+        Parameters
+        ----------
+        collapsed : `numpy.ma.masked_array`
+            Masked array to check the edges of.
+
+        Returns
+        -------
+        maskArray : `list` [`bool`]
+            List of pixels to mask.
+        """
+        maskArray = [False for element in collapsed]
+        if np.ma.is_masked(collapsed):
+            num = len(collapsed)
+            for low in range(num):
+                if not collapsed.mask[low]:
+                    break
+            if low > 0:
+                maskArray[:low] = True
+            for high in range(1, num):
+                if not collapsed.mask[-high]:
+                    break
+            if high > 1:
+                maskArray[-high:] = True
+        return maskArray
+
+    def overscanVector(self, image, order):
+        """Calculate the 1-d vector overscan from the input overscan image.
+
+        Parameters
+        ----------
+        image : `lsst.afw.image.maskedImage`
+            Image containing the overscan data.
+        order : `int`
+            Fit order to use for polynomials; number of spline points
+            to use for evaluation.
+
+        Returns
+        -------
+        overscanVector : `numpy.ndarray`
+            Vector containing the overscan solution.
+        maskArray : `list` [`bool`]
+            List of rows that should be masked as SUSPECT when the
+            overscan solution is applied.
+        isTransposed : `bool`
+            Indicates if the overscan data was transposed during
+            calcuation, noting along which axis the overscan should be
+            subtracted.
+        """
+        calcImage = self.getCalcImage(image)
+
+        # operate on numpy-arrays from here
+        calcImage, isTransposed = self.transpose(calcImage)
+        masked = self.maskOutliers(calcImage)
+
+        if self.fitType == 'MEDIAN_PER_ROW':
+            overscanVector = self.collapseArrayMedian(masked)
+            maskArray = self.extrapolationMask(overscanVector)
+        else:
+            collapsed = self.collapseArray(masked)
+
+            num = len(collapsed)
+            indices = 2.0*np.arange(num)/float(num) - 1.0
+
+            poly = np.polynomial
+            fitter, evaler = {
+                'POLY': (poly.polynomial.polyfit, poly.polynomial.polyval),
+                'CHEB': (poly.chebyshev.chebfit, poly.chebyshev.chebval),
+                'LEG': (poly.legendre.legfit, poly.legendre.legval),
+                'NATURAL_SPLINE': (self.splineFit, self.splineEval),
+                'CUBIC_SPLINE': (self.splineFit, self.splineEval),
+                'AKIMA_SPLINE': (self.splineFit, self.splineEval)
+            }[self.fitType]
+
+            coeffs = fitter(indices, collapsed, order)
+            overscanVector = evaler(indices, coeffs)
+            maskArray = self.extrapolationMask(collapsed)
+        return overscanVector, maskArray, isTransposed
+
+    def debugView(self, image, model):
```

Comment:
I think the plotting should be called from `OverscanCorrectionTask.run`, rather than as an optional extra method.